src/sdam/message_manager.rs

```diff
@@ -0,0 +1,82 @@
+use std::{sync::Arc, time::Duration};
+
+use futures::future::Either;
+use futures_timer::Delay;
+use tokio::sync::watch::{self, Receiver, Sender};
+
+/// Provides functionality for message passing between server selection operations and SDAM
+/// background tasks.
+#[derive(Clone, Debug)]
+pub(crate) struct TopologyMessageManager {
+    topology_check_requester: Arc<Sender<()>>,
+    topology_check_listener: Receiver<()>,
+    topology_change_notifier: Arc<Sender<()>>,
+    topology_change_listener: Receiver<()>,
+}
+
+impl TopologyMessageManager {
+    /// Constructs a new TopologyMessageManager.
+    pub(super) fn new() -> Self {
+        let (topology_check_requester, topology_check_listener) = watch::channel(());
+        let (topology_change_notifier, topology_change_listener) = watch::channel(());
+
+        Self {
+            topology_check_requester: Arc::new(topology_check_requester),
+            topology_check_listener,
+            topology_change_notifier: Arc::new(topology_change_notifier),
+            topology_change_listener,
+        }
+    }
+
+    /// Gets a receiver which will receive messages whenever a topology check has been requested by
+    /// a server selection operation. This should be called by each SDAM background task when it
+    /// starts up.
+    pub(super) fn subscribe_to_topology_check_requests(&self) -> Receiver<()> {
```

Comment:
Do we need this method? I imagine the SDAM background tasks could just use `wait_for_topology_check_request`.

Reply:
Nope, we can delete it.